python/taichi/lang/sparse_matrix.py

```diff
@@ -39,9 +39,15 @@ def transpose(self):
         return SparseMatrix(sm=sm)
 
     def __matmul__(self, other):
-        assert self.m == other.n, f"Dimension mismatch between sparse matrices ({self.n}, {self.m}) and ({other.n}, {other.m})"
-        sm = self.matrix.matmul(other.matrix)
-        return SparseMatrix(sm=sm)
+        import numpy as np
+        if isinstance(other, SparseMatrix):
+            assert self.m == other.n, f"Dimension mismatch between sparse matrices ({self.n}, {self.m}) and ({other.n}, {other.m})"
+            sm = self.matrix.matmul(other.matrix)
+            return SparseMatrix(sm=sm)
+        elif isinstance(other, np.ndarray):
+            assert self.m == other.shape[
+                0], f"Dimension mismatch between sparse matrix ({self.n}, {self.m}) and vector ({other.shape})"
+            return self.matrix.mat_vec_mul(other)
```

Comment:
You may want to call `.to_numpy()` here so that your users don't have to when calling `A @ b`.